param/__init__.py

```diff
@@ -132,12 +133,20 @@ def named_objs(objlist, namesdict=None):
     """
     objs = OrderedDict()
 
+    objtoname = {}
+    unhashables = []
     if namesdict is not None:
-        objtoname = {hashable(v): k for k, v in namesdict.items()}
+        for k, v in namesdict.items():
+            try:
+                objtoname[hashable(v)] = k
+            except TypeError:
+                unhashables.append((k, v))
 
     for obj in objlist:
-        if namesdict is not None and hashable(obj) in objtoname:
+        if objtoname and hashable(obj) in objtoname:
             k = objtoname[hashable(obj)]
+        elif any(obj is v for (_, v) in unhashables):
+            k = [k for (k, v) in unhashables if v is obj][0]
```

Comment:
This seems to be handling the case where the same unhashable object is in the list multiple times with different names, but then it returns only the first name?  If so what's the use of `unhashables` being a list rather than a dict?

Reply:
Well it can't be a dict since the obj is unhashable.

Reply:
A k->v dict.

Reply:
Ah, I guess that would make for a nicer data structure but then awkward to use. Never mind.